src/lightning_app/core/work.py

```diff
@@ -226,8 +231,17 @@ def cloud_compute(self) -> CloudCompute:
         return self._cloud_compute
 
     @cloud_compute.setter
-    def cloud_compute(self, cloud_compute) -> None:
+    def cloud_compute(self, cloud_compute: CloudCompute) -> None:
         """Returns the cloud compute used to select the cloud hardware."""
+        # A new ID
+        current_id = self._cloud_compute.id
+        new_id = cloud_compute.id
+        if current_id != new_id:
+            compute_store: _CloudComputeStore = _CLOUD_COMPUTE_STORE[current_id]
+            if compute_store.frozen:
+                raise Exception("The current cloud compute has already been frozen and can't be changed anymore.")
```

Comment:
I’m not sure developers would know what “freezing” is and why we freeze. We can return a more informative exception message that doesn’t rely on our jargon.

Reply:
I removed it as this isn't required anymore with dynamic ports suppport.